radix-engine/src/errors.rs

```diff
@@ -411,6 +422,7 @@ pub enum ApplicationError {
     // TODO: this should never happen because of schema check?
     InputDecodeError(DecodeError),
 
+    /// A panic was encountered in Scrypto code.
```

Comment:
Minor: It can be used for by both native and Scrypto. Maybe we should rename `Panic` to `PanicMessage`?

Reply:
Sounds good!